File: Aran/RNAV/Libs/Panda/Geometry/C#/MultiPoint.cs
using System;
using System.Collections.Generic;
using System.Text;
using ARAN.Common;
using ARAN.Contracts.Registry;
using System.Runtime.InteropServices;

namespace ARAN.GeometryClasses
{
	public class MultiPoint : GeometryList<Point>
	{
		public MultiPoint()
			: base()
		{

		}

		public override GeometryType GetGeometryType()
		{
			return GeometryType.MultiPoint;
		}

		protected Point GetPoint(int index)
		{
			return base[index];
		}

		public void AddPoint(Point point)
		{
			Add(point);
		}

		public void AddMultiPoint(MultiPoint multiPoint)
		{
			int i;
			for (i = 0; i <= multiPoint.Count - 1; i++)
				Add(multiPoint.GetPoint(i));
		}

		public void InsertPoint(int index, Point point)
		{
			Insert(index, point);
		}


		public override Object Clone()
		{
			MultiPoint mp = new MultiPoint();
			mp.Assign(this);
			return mp;
		}

		/* public override void Assign (PandaItem source)
		 {
			 int i;
			 MultiPoint dst;
			 dst = (MultiPoint)source;
			 dst.Clear();
			 for (i = 0; i <= GetCount - 1; i++)
				 dst.AddPoint(GetPoint(i));
		 }
		 */


		public override Point this[int index]
		{
			get
			{
				return GetPoint(index);
			}

		}
		
}

}

File: Aran/RNAV/Libs/Panda/Geometry/cpp/src/MultiPoint.cpp
#include "../include/MultiPoint.h"
#include "../include/Point.h"
#include "../include/GeometryType.h"

namespace Panda
{
	MultiPoint::MultiPoint() throw ()
	{
	}

	MultiPoint::~MultiPoint() throw ()
	{
		clear ();
	}

	void MultiPoint::addPoint (const Point& point)
	{
		_pointList.add (point);
	}

	void MultiPoint::insertPoint (int index, const Point& point)
	{
		_pointList.insert (index, point);
	}
	 
	const Point& MultiPoint::getPoint (int index) const
	{
		return _pointList.at (index);
	}

	void MultiPoint::removeAt ( int index)
	{
		_pointList.remove (index);
	}

	int MultiPoint::count() const
	{
		return _pointList.size ();
	}

	void MultiPoint::clear ()
	{
		_pointList.clear ();
	}

	Geometry* MultiPoint::clone() const throw (std::bad_cast)
	{
		MultiPoint* multiPoint = new MultiPoint ();
		try
		{
			multiPoint->assign (*this);
		}
		catch (std::bad_cast e)
		{
			delete multiPoint;
			throw e;
		}

		return multiPoint;
	}

	void MultiPoint::assign (const Geometry& geometry) throw (std::bad_cast)
	{
		const MultiPoint& src = dynamic_cast <const MultiPoint&> (geometry);
		_pointList.assign (src._pointList);
	}

	void MultiPoint::pack (Handle handle) const throw (Registry::Exception)
	{
		_pointList.pack (handle);
	}

	void MultiPoint::unpack (Handle handle) throw (Registry::Exception)
	{
		_pointList.unpack (handle);
	}

	int MultiPoint::geometryType () const throw ()
	{
		return GeometryType::MultiPoint;
	}
}
